### ISSUE TITLE
Multiple CacheEntry objects with same key but different expires on DateTime

### ISSUE THREAD

**Issue by kambg** [openrasta/openrasta-caching#6]
In CachingContributor it looks like we always add CacheEntry to the store as long as we have a GET request method and status of 200.
Having this in mind we can end up with multiple CacheEntry objects with same key but different expires on DateTime.
What is the purpose of this mechanism, I would've thought we'd want to have single CacheEntry per key and that to be the latest resource that satisfied a request?

Cheers

**Comment by serialseb** [comment]
Nope. The cache key is on the URI _plus_ some headers on which the response to the cache vary, so you may have multiple cached representations for the same http header.

Don't remember the specifics but am pretty sure that's why it was built that way

## 

Seb

---

From: Kamen84 [reply@reply.github.com]
Sent: 05 January 2012 17:51
To: Sebastien Lambla
Subject: [openrasta-caching] Multiple CacheEntry objects with same key but different expires on DateTime (#6)

In CachingContributor it looks like we always add CacheEntry to the store as long as we have a GET request method and status of 200.
Having this in mind we can end up with multiple CacheEntry objects with same key but different expires on DateTime.
What is the purpose of this mechanism, I would've thought we'd want to have single CacheEntry per key and that to be the latest resource that satisfied a request?

Cheers

---

Reply to this email directly or view it on GitHub:
https://github.com/openrasta/openrasta-caching/issues/6

**Comment by kambg** [comment]
Thanks Seb. In the current implementation it looks like the VaryHeaders are always an empty dictionary. Can you give an example of what VaryHeaders should contain?
